### PR TITLE
feat(iOS): rewrite old arch to use UIScrollView

### PR DESCRIPTION
**Pull request by okwasniewski**
# Summary

This PR rewrites old architecture of react-native-pager-view to UIScrollView on iOS. 

## Test Plan

Launch the example app.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ❌     |

## Checklist

- [X] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I updated the typed files (TS and Flow)

## Known issues
- [ ] Swiping right on last page is stuck on settling when overdrag is disabled.

### CODE REVIEW

File: ios/ReactNativePageView.m [hunk 1]
```diff
@@ -127,338 +93,88 @@ - (void)shouldDismissKeyboard:(NSString *)dismissKeyboard {
...
-    } else {
-        [self goTo:newIndex animated:NO];
+    if (self.initialPage != 0) {
+        [self goTo:self.initialPage animated:false];
     }
 }
 
 - (void)disableSwipe {
-    self.reactPageViewController.view.userInteractionEnabled = NO;
+    self.scrollView.userInteractionEnabled = NO;
```
[the remark this answers is not in the document]
Comment (okwasniewski) [reply ios/ReactNativePageView.m]: You are right, changed it 👍🏻

File: ios/RNCPagerView.h [hunk 2]
```diff
@@ -5,23 +5,19 @@
...
 
-- (instancetype)initWithEventDispatcher:(RCTEventDispatcher *)eventDispatcher;
+- (instancetype)initWithEventDispatcher:(id<RCTEventDispatcherProtocol> )eventDispatcher;
 
 @property(nonatomic) NSInteger initialPage;
-@property(nonatomic) NSInteger lastReportedIndex;
-@property(nonatomic) NSInteger destinationIndex;
+@property(nonatomic) NSString* orientation;
 @property(nonatomic) NSInteger currentIndex;
 @property(nonatomic) NSInteger pageMargin;
```
Comment (krozniata) [inline ios/RNCPagerView.h:15]: I think this can be removed, because it's not used anymore inside `RNCPagerView.m`

File: ios/RNCPagerView.m [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+#import <math.h>
+
+@interface RNCPagerView () <UIScrollViewDelegate>
+
+@property(nonatomic, strong) id<RCTEventDispatcherProtocol> eventDispatcher;
+
+@property(nonatomic, strong) UIScrollView *scrollView;
+@property(nonatomic, strong) UIView *containerView;
+
+@property(nonatomic, assign) CGPoint lastContentOffset;
```
Comment (krozniata) [inline ios/RNCPagerView.m:20]: Why do we need this property?

**Review by krozniata — APPROVED** [review]

**Comment by andreialecu** [comment]
I'm curious about how this is different from a regular RN ScrollView with `pagingEnabled: true`?

I had issues with RN ScrollView in the past and moved to `react-native-pager-view` specifically to solve those.

Issues were: it would randomly get stuck between pages at some points, and performance seemed worse in general, especially with Reanimated and complex interactions.

**Comment by okwasniewski** [comment]
Hey @andreialecu!   

There are quite a lot of differences between those two implementations. The Core RN ScrollView is a generic scrolling component written to support as many use cases as possible to be the “go-to” scrolling component. 
 
Both implementations are based on the native `UIScrollView` component, but pager's implementation is focused on providing the best paging experience possible, while keeping the same API as old implementation. You can read more about the motivation here: https://github.com/callstack/react-native-pager-view/issues/673
 
Also, we are going to soon release pager-view `7.0.0-next` it would be awesome If you could test it out on your specific use case and let us know if you have any performance issues.

**Pull request merged**